packages/sdk/src/actions/prepareAtomicTx.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  Hex,
+  // PublicClient,
+  WalletClient,
+  parseAbi,
+  // SimulateCallsReturnType,
+  Call,
+  encodeFunctionData,
+  concatHex,
+} from "viem";
+import { Quote } from "./getQuote.js";
+import { LoggerT } from "../utils/index.js";
+import { spokePoolAbiV3_5 } from "../abis/SpokePool/index.js";
+import { addressToBytes32, getIntegratorDataSuffix } from "../utils/index.js";
+
+export type PrepareAtomicTxParams = {
+  walletClient: WalletClient;
+  // publicClient: PublicClient;
+  deposit: Quote["deposit"];
+  approvalAmount: bigint;
+  integratorId: Hex;
+  logger?: LoggerT;
+};
+
+export type PrepareAtomicTxResult = {
+  // simulateResult?: SimulateCallsReturnType<readonly Call[]>;
+  calls: readonly Call[];
+  error?: Error;
+};
+
+export async function prepareAtomicTx(
```

Comment:
NIT: Based on the naming, I would assume that this function can prepare any atomic tx. A more concise naming like `prepareAtomicApproveDepositTx` could make sense.